lib/helpers/date.dart

```diff
@@ -30,6 +30,51 @@ List<DateTime> daysInRange(DateTime first, DateTime last) {
   );
 }
 
+/// Formats a Duration into a human-readable string with years, months, and days
+String humanDuration(DateTime startDate, DateTime endDate) {
```

Comment:
I would just use an external package for this (one of the very first results was https://pub.dev/packages/humanize_duration), then we would also have i18n solved

Reply:
these types of libraries don't do it the way i think makes sense.
e.g. day A you reported a weight at 9:20 AM, one day later you weigh in at 9:00 AM, i think that should qualify as "a day"